web-api/src/personalization/routes.rs

```diff
@@ -472,20 +478,68 @@ impl SemanticSearchQuery {
     }
 }
 
+/// Represents body of a POST `semantic_search_from_text` request.
+#[derive(Debug, Clone, Deserialize)]
+struct SemanticSearchText {
+    text: String,
+    document_count: Option<usize>,
```

Comment:
🔧 see the comments in the spec file
```suggestion
struct SemanticSearchRequest {
    text: String,
    count: Option<usize>,
```

Reply:
I think SemanticSearchRequest is already taken, but will check

Reply:
for the GET we use `SemanticSearchQuery`, so it would fit well to use `SemanticSearchRequest` for the POST